include/veriblock/blockchain/commands/addblock.hpp

```diff
@@ -35,12 +35,12 @@ struct AddBlock : public Command {
   void UnExecute() override {
     auto hash = block_->getHash();
     auto* index = tree_->getBlockIndex(hash);
-    if (!index) {
-      return;
-    }
+    assert(index != nullptr
```

Comment:
UnExecute is not guaranteed to be executed after command applied. It will be executed on AddBlock if it failed. And if so, this block will not exist in blockchain, so nothing to remove here. `index` will be nullptr here.

Reply:
Reference counting blocks only makes sense if Execute and UnExecute calls match.Otherwise it becomes possible for example to execute a command group that adds blocks 1,2,3(eg the BTC context of a VTB), and then unexecute a different command group that adds blocks 1 and 2(the context shipped by another VTB)